From 1bb9ebaf5d3ed11587d0f6715b113dd72a7663a6 Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Mon, 9 Oct 2017 16:25:33 +0800
Subject: [PATCH 2648/5242] MLK-16586-1 clk: imx8qm: add the cm41 ipg clk

commit  db4c578a142219196112ff2f584498202a1c751a from
https://source.codeaurora.org/external/imx/linux-imx.git

Add the cm41 ipg clk

BuildInfo:
- SCFW a6fd9a48, IMX-MKIMAGE 0, ATF 0
- U-Boot 2017.03-imx_v2017.03_4.9.11_imx8_alpha+g258936c

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Reviewed-by: Robin Gong <yibin.gong@nxp.com>
Reviewed-by: Dong Aisheng <aisheng.dong@nxp.com>
Tested-by: Andy Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx8qm.c             |    1 +
 include/dt-bindings/clock/imx8qm-clock.h |    3 ++-
 include/dt-bindings/soc/imx8_pd.h        |    2 ++
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/clk/imx/clk-imx8qm.c b/drivers/clk/imx/clk-imx8qm.c
index 97c1b29..a99176b 100644
--- a/drivers/clk/imx/clk-imx8qm.c
+++ b/drivers/clk/imx/clk-imx8qm.c
@@ -578,6 +578,7 @@ static int imx8qm_clk_probe(struct platform_device *pdev)
 	clks[IMX8QM_SAI6_RX_BCLK]          = imx_clk_fixed("sai6_rx_bclk", 0);
 	clks[IMX8QM_HDMI_TX_SAI0_TX_BCLK]  = imx_clk_fixed("hdmi_tx_sai0_tx_bclk", 0);
 	clks[IMX8QM_CM40_IPG_CLK]	= imx_clk_fixed("ipg_cm40_clk_root", SC_132MHZ);
+	clks[IMX8QM_CM41_IPG_CLK]	= imx_clk_fixed("ipg_cm41_clk_root", SC_132MHZ);
 
 	np_acm = of_find_compatible_node(NULL, NULL, "nxp,imx8qm-acm");
 	if (np_acm) {
diff --git a/include/dt-bindings/clock/imx8qm-clock.h b/include/dt-bindings/clock/imx8qm-clock.h
index a5b57c4..bd0db40 100644
--- a/include/dt-bindings/clock/imx8qm-clock.h
+++ b/include/dt-bindings/clock/imx8qm-clock.h
@@ -825,7 +825,8 @@
 
 /* CM40 */
 #define IMX8QM_CM40_IPG_CLK				772
+#define IMX8QM_CM41_IPG_CLK				773
 
-#define IMX8QM_CLK_END					773
+#define IMX8QM_CLK_END					774
 
 #endif /* __DT_BINDINGS_CLOCK_IMX8QM_H */
diff --git a/include/dt-bindings/soc/imx8_pd.h b/include/dt-bindings/soc/imx8_pd.h
index 1483ac3..5727506 100644
--- a/include/dt-bindings/soc/imx8_pd.h
+++ b/include/dt-bindings/soc/imx8_pd.h
@@ -190,6 +190,8 @@
 #define PD_CM40                     cm40_power_domain
 #define PD_CM40_I2C                 cm40_i2c
 #define PD_CM40_INTMUX              cm40_intmux
+#define PD_CM41                     cm41_power_domain
+#define PD_CM41_INTMUX              cm41_intmux
 
 #endif /* __DT_BINDINGS_IMX8_PD_H */
 
-- 
1.7.9.5

